polaris/dataset/converters/_pdb.py

```diff
@@ -0,0 +1,158 @@
+        """
+        pdb_dict = {k: [] for k in KEYS}
+        for row in atom_array:
+            pdb_dict["X"].append(row.coord[0])
+            pdb_dict["Y"].append(row.coord[1])
+            pdb_dict["Z"].append(row.coord[2])
+            pdb_dict["chain_id"].append(row.chain_id)
+            pdb_dict["res_id"].append(row.res_id)
+            pdb_dict["ins_code"].append(row.ins_code)
+            pdb_dict["res_name"].append(row.res_name)
+            pdb_dict["hetero"].append(row.hetero)
+            pdb_dict["atom_name"].append(row.atom_name)
+            pdb_dict["element"].append(row.element)
+            pdb_dict["atom_id"].append(row.atom_id)
+            pdb_dict["b_factor"].append(row.b_factor)
+            pdb_dict["occupancy"].append(row.occupancy)
+            pdb_dict["charge"].append(row.charge)
+        return pdb_dict
+
+    def _load_pdf(self, path: str, pdb_pointer=None) -> dict:
+        """
+        Load a single PDB file along with a dictionary that contains structure properties as keys.
+
+        `AtomArray` stores data for an entire structure model containing *n* atoms.
+        See [AtomArray](https://github.com/biotite-dev/biotite/blob/0404084283765baef765cc869d86ee07a898d82f/src/biotite/structure/atoms.py#L558) for more details.
+        """
+        # load pdb file
+        in_file = fastpdb.PDBFile.read(path)
+
+        # get the structure as AtomArray
+        atom_array = in_file.get_structure(
+            model=1, include_bonds=True, extra_fields=["atom_id", "b_factor", "occupancy", "charge"]
+        )
+
+        # convert AtomArray to dict
+        pdb_dict = self._pdb_to_dict(atom_array)
+
+        return pdb_dict
+
+    def _convert_pdb(
+        self, path: str, factory: "DatasetFactory", pdb_pointer: Union[str, int]
+    ) -> FactoryProduct:
+        """
+        Convert a single pdb to zarr file
+        """
+
+        # load pdb as fastpbd and convert to dict
+        pdb_dict = self._load_pdf(path)
+
+        # Create group and add datasets
+        if self.pdb_column not in factory.zarr_root:
+            pdb_group = factory.zarr_root.create_group(self.pdb_column)
+        else:
+            pdb_group = factory.zarr_root[self.pdb_column]
+
+        group = pdb_group.create_group(pdb_pointer)
+        for col_name, col_val in pdb_dict.items():
+            col_val = np.array(col_val)
+            # get the accepted dtype
+            dtype = DTYPE_DICT.get(col_name, col_val.dtype)
+            group.create_dataset(col_name, data=col_val, dtype=dtype)
+
+    def convert(self, path: Union[str, List[str]], factory: "DatasetFactory") -> FactoryProduct:
+        """Convert one or a list of PDB files into Zarr"""
+
+        if not isinstance(path, list):
+            path = [path]
+
+        pdb_pointers = []
+
+        # load single pdb and convert to zarr
+        for pdb_path in path:
+            # use the pdb file name as pointer
+            pdb_pointer = Path(pdb_path).stem
```

Comment:
**Suggestion:** Using `Path(...).stem` as pointer may lead to duplicate pointers. I think conversion should fail in that case. I think this is an issue for the other converters as well.